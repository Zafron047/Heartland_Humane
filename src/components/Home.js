import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { fetchKatBreeds } from '../redux/homeReducer/HomeSlice';
import '../styles/Home.css';
import arrow from '../assets/arrowRight.svg';
import cat from '../assets/cat.jpg';

const Home = () => {
  const breedList = useSelector((state) => state.AllKats.breedList);
  const loading = useSelector((state) => state.AllKats.loading);
  const error = useSelector((state) => state.AllKats.error);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchKatBreeds());
  }, [dispatch]);

  const [searchQuery, setSearchQuery] = useState('');

  const handleSearchInputChange = (event) => {
    setSearchQuery(event.target.value);
  };

  const filteredBreeds = breedList.filter((breed) => breed.name
    .toLowerCase().includes(searchQuery.toLowerCase()));

  let content;

  if (loading) {
    content = <p className="home-loading">Loading...</p>;
  } else if (error) {
    content = <p>{error}</p>;
  } else {
    content = (
      <>
        <input
          className="search-bar-wrapper"
          type="text"
          placeholder="Search by breed name..."
          value={searchQuery}
          onChange={handleSearchInputChange}
        />
        <ul className="table">
          {filteredBreeds.map((breed) => (
            <Link className="table-row" key={breed.id} to={`/Details/${breed.id}`}>
              <li className="table-cell">
                <img className="arrow" alt="arrow" src={arrow} />
                <div className="left-wing">
                  <h2 className="breed-name">
                    {breed.name}
                  </h2>
                  <p className="feature">
                    Origin:
                    {' '}
                    {breed.origin}
                  </p>
                </div>
              </li>
            </Link>
          ))}
        </ul>
      </>
    );
  }

  return (
    <div className="home-container">
      <div className="header">
        <img className="header-img" src={cat} alt="Header" />
      </div>
      {content}
    </div>
  );
};

export default Home;
