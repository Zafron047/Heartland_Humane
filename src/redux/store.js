import { configureStore } from '@reduxjs/toolkit';
import homeReducer from './homeReducer/HomeSlice';
import detailReducer from './detailReducer/detailSlice';

const store = configureStore({
  reducer: {
    AllKats: homeReducer,
    CatBreeds: detailReducer,
  },
});

export default store;
