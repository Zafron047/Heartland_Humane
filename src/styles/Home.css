@media only screen and (max-width: 767px) {
  .home-container {
    background-color: rgba(205, 44, 125, 0.819);
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    align-items: center;
  }

  .header {
    display: flex;
  }

  .header-img {
    width: 90vw;
    border: 3px solid aliceblue;
    border-radius: 30px;
    opacity: 0.8;
    margin: 1.5rem 0;
  }

  .headline {
    text-decoration: underline;
    text-underline-offset: 10px;
    color: aliceblue;
  }

  .search-bar-wrapper {
    width: 100%;
    margin: 0 auto;
  }

  input[type="text"] {
    width: 80%;
    padding: 10px;
    margin-bottom: 10px;
    font-size: 16px;
    border: 1px solid #ccc;
    border-radius: 4px;
  }

  .table-wrapper {
    margin-top: 10px;
  }

  .Breeds {
    color: purple;
    margin: 0;
    padding: 0;
    align-self: flex-start;
  }

  .temp {
    display: flex;
    flex-direction: column;
  }

  .table {
    display: flex;
    flex-wrap: wrap;
    gap: 15px;
    padding: 0;
    justify-content: center;
  }

  .table a {
    text-decoration: none;
    color: purple;
  }

  .table-cell {
    position: relative;
    background-color: rgb(255, 105, 180);
    border-radius: 10px;
    border-bottom: 1px solid #ccc;
    text-align: center;
    width: 10rem;
    height: 10rem;
  }

  .arrow {
    position: absolute;
    top: 5px;
    right: 5px;
    opacity: 0.6;
  }

  .left-wing {
    position: absolute;
    bottom: 10px;
    width: 90%;
    left: 50%;
    transform: translateX(-50%);
  }

  .breed-name {
    font-size: 1.4rem;
  }

  .feature {
    font-size: small;
  }

  .home-loading {
    display: flex;
    justify-content: center;
    align-items: center;
    color: aliceblue;
    font-size: 12vw;
  }
}
