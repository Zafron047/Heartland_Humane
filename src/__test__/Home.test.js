import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Home from '../components/Home';

const customRender = (component) => render(
  <Provider store={store}>
    {component}
  </Provider>,
);

describe('Home', () => {
  it('render Home Components', () => {
    customRender(<Home />);
    expect(Home).toMatchSnapshot();
  });
});
