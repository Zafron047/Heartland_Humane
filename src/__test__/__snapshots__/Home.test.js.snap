// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home render Home Components 1`] = `[Function]`;
